### ISSUE TITLE
Ordering of JSX spread attributes

### ISSUE THREAD

**Issue by theetrain** [yannickcr/eslint-plugin-react#1602]
Hello, not sure if this was discussed already. In my project, I use '[spread attributes](https://reactpatterns.com/#jsx-spread-attributes)' a certain way:

```jsx
const Link = ({ children, ...rest }) => (
  // `rest` attributes are spread at the beginning
  // so that `className` and other props can override user-submitted props
  <a {...rest} className="link">{children}</a>
)
```

Is there currently a way to enforce this order so that something like this would throw an error:

```jsx
const Link = ({ children, ...rest }) => (
  // error, incorrect ordering of spread attributes
  <a className="link" {...rest}>{children}</a>
)
```

**Comment by ljharb** [comment]
No, there's not - and a rule like that would be a very bad idea. The correct ordering of spread props is very context-specific - in your first example, the inline `className` overrides a passed-in `className`; in your second one, the inline one overrides it.

Which of those is correct is always entirely dependent on your use case, so it wouldn't make sense to enforce that with a linting rule. That's something unit tests should cover.

**Comment by theetrain** [comment]
@ljharb indeed it depends on the use case for an individual component. I don't mean to say there is a 'right way', just that it would be helpful to detect which method the developer prefers. Maybe something like `jsx-spread-props: "before"`.

Within my project, I always want my props to override the user's passed-in props as they consume the component. I understand your point though; making this a linting rule would change the overall behaviour of this situation every time it comes up. I appreciate that insight.

**Comment by ljharb** [comment]
Linter rules apply to projects, across files - this kind of thing would only apply per *call site*, not even for an entire component.
